src/main/java/seedu/address/logic/commands/EditContactCommand.java

```diff
@@ -0,0 +1,215 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.core.Messages.MESSAGE_CONTACT_NAME_DOES_NOT_EXIST;
+import static seedu.address.commons.core.Messages.MESSAGE_DUPLICATE_CONTACT;
+import static seedu.address.commons.core.Messages.MESSAGE_EDIT_CONTACT_SUCCESS;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_EMAIL;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_NAME;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_PHONE;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_TAG;
+import static seedu.address.model.Model.PREDICATE_SHOW_ALL_CONTACTS;
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+import seedu.address.commons.util.CollectionUtil;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.contact.Contact;
+import seedu.address.model.contact.ContactEmail;
+import seedu.address.model.contact.ContactName;
+import seedu.address.model.contact.ContactPhone;
+import seedu.address.model.tag.Tag;
+
+/**
+ * Edits the details of an existing contact in Teaching Assistant.
+ */
+public class EditContactCommand extends Command {
+
+    public static final String COMMAND_WORD = "cedit";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Edits the details of the contact identified "
+            + "by name used in the displayed contact list. "
+            + "Existing values will be overwritten by the input values.\n"
+            + "Parameters: NAME "
```

Comment:
is this going to be a problem when there are two contacts of the same name but different phone numbers? because that's what i thought